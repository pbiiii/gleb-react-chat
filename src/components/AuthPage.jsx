import React from 'react';
import { Redirect } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import AuthForm from './AuthForm';
import ErrorMessage from './ErrorMessage';

const styles = theme => ({
    paper: {
        marginTop: theme.spacing.unit * 3 + 64,
        width: 500,
    },
    tabContent: {
        padding: theme.spacing.unit * 3,
    },
});

class AuthPage extends React.Component {
    state = {
        activeTab: 0,
        form: {
            username: {
                value: '',
                isValid: true,
            },
            password: {
                value: '',
                isValid: true,
            },
            repeatedPassword: {
                value: '',
                isValid: true,
            },
        },
    };

    handleTabChange = (event, value) => {
        this.setState({ activeTab: value });
    };

    onInputChange = (event) => {
        event.preventDefault();
        const { name, value } = event.target;
        this.setState(prevState => ({
            form: {
                ...prevState.form,
                [name]: {
                    ...prevState.form[name],
                    value,
                },
            },
        }));
    };

    handleRegisterSubmit = (event) => {
        event.preventDefault();
        const { form: { username, password } } = this.state;
        const { register } = this.props;
        register({
            username: username.value,
            password: password.value,
        });
    };

    handleLoginSubmit = (event) => {
        event.preventDefault();
        const { form: { username, password } } = this.state;
        const { login } = this.props;
        login({
            username: username.value,
            password: password.value,
        });
    };

    render() {
        const {
            classes, isAuthenticated, error,
        } = this.props;

        const { activeTab, form } = this.state;

        if (isAuthenticated) {
            return <Redirect to="/chat" />;
        }

        return (
            <React.Fragment>
                <AppBar>
                    <Toolbar>
                        <Typography variant="title" color="inherit" style={{ flex: 1 }}>
                            Gleb React Chat
                        </Typography>
                    </Toolbar>
                </AppBar>
                <Grid container justify="center">
                    <Grid item>
                        <Paper className={classes.paper}>
                            <AppBar position="static" color="default">
                                <Tabs value={activeTab} onChange={this.handleTabChange} fullWidth>
                                    <Tab label="Login" />
                                    <Tab label="Register" />
                                </Tabs>
                            </AppBar>
                            <div className={classes.tabContent}>
                                {activeTab === 0 && (
                                    <AuthForm
                                        onSubmit={this.handleLoginSubmit}
                                        onChange={this.onInputChange}
                                        {...form}
                                    />
                                )}
                                {activeTab === 1 && (
                                    <AuthForm
                                        onSubmit={this.handleRegisterSubmit}
                                        onChange={this.onInputChange}
                                        {...form}
                                        isRegisterForm={true}
                                    />
                                )}
                            </div>
                        </Paper>
                    </Grid>
                </Grid>
                <ErrorMessage error={error} />
            </React.Fragment>
        );
    }
}

export default withStyles(styles)(AuthPage);
