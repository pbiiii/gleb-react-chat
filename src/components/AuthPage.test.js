/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AuthPage from './AuthPage';

const mockProps = {
    isAuthenticated: false,
    register: () => {},
    login: () => {},
}

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AuthPage {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<AuthPage {...mockProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
