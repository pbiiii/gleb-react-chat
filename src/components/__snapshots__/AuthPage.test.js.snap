// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-298 MuiPaper-elevation4-304 MuiAppBar-root-289 MuiAppBar-positionFixed-290 MuiAppBar-colorPrimary-296 mui-fixed"
  >
    <div
      className="MuiToolbar-root-325 MuiToolbar-regular-327 MuiToolbar-gutters-326"
    >
      <h2
        className="MuiTypography-root-329 MuiTypography-title-335 MuiTypography-colorInherit-358"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        Gleb React Chat
      </h2>
    </div>
  </header>,
  <div
    className="MuiGrid-container-364 MuiGrid-justify-xs-center-381"
  >
    <div
      className="MuiGrid-item-365"
    >
      <div
        className="MuiPaper-root-298 MuiPaper-elevation2-302 MuiPaper-rounded-299 AuthPage-paper-287"
      >
        <header
          className="MuiPaper-root-298 MuiPaper-elevation4-304 MuiAppBar-root-289 MuiAppBar-positionStatic-293 MuiAppBar-colorDefault-295"
        >
          <div
            className="MuiTabs-root-461"
          >
            <div
              className="MuiTabs-flexContainer-462"
            >
              <div
                className="MuiTabs-scroller-464 MuiTabs-fixed-465"
                onScroll={[Function]}
                role="tablist"
                style={
                  Object {
                    "marginBottom": 0,
                  }
                }
              >
                <div
                  className="MuiTabs-flexContainer-462"
                >
                  <button
                    aria-selected={true}
                    className="MuiButtonBase-root-482 MuiTab-root-470 MuiTab-textColorInherit-472 MuiTab-selected-475 MuiTab-fullWidth-477"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper-478"
                    >
                      <span
                        className="MuiTab-labelContainer-479"
                      >
                        <span
                          className="MuiTab-label-480"
                        >
                          Login
                        </span>
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-566"
                    />
                  </button>
                  <button
                    aria-selected={false}
                    className="MuiButtonBase-root-482 MuiTab-root-470 MuiTab-textColorInherit-472 MuiTab-fullWidth-477"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper-478"
                    >
                      <span
                        className="MuiTab-labelContainer-479"
                      >
                        <span
                          className="MuiTab-label-480"
                        >
                          Register
                        </span>
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-566"
                    />
                  </button>
                </div>
                <span
                  className="MuiPrivateTabIndicator-root-485 MuiPrivateTabIndicator-colorSecondary-487 MuiTabs-indicator-469"
                  style={
                    Object {
                      "left": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </div>
        </header>
        <div
          className="AuthPage-tabContent-288"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="MuiFormControl-root-488 MuiFormControl-marginNormal-489 MuiFormControl-fullWidth-491"
            >
              <label
                className="MuiFormLabel-root-503 MuiFormLabel-required-508 MuiInputLabel-required-496 MuiInputLabel-root-492 MuiInputLabel-formControl-497 MuiInputLabel-animated-500"
                data-shrink={false}
              >
                Username
                <span
                  className="MuiFormLabel-asterisk-509"
                >
                   *
                </span>
              </label>
              <div
                className="MuiInputBase-root-523 MuiInput-root-510 MuiInput-underline-514 MuiInputBase-fullWidth-532 MuiInput-fullWidth-517 MuiInputBase-formControl-524 MuiInput-formControl-511"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="username"
                  className="MuiInputBase-input-533 MuiInput-input-518"
                  disabled={false}
                  name="username"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Type your username..."
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="MuiFormControl-root-488 MuiFormControl-marginNormal-489 MuiFormControl-fullWidth-491"
            >
              <label
                className="MuiFormLabel-root-503 MuiFormLabel-required-508 MuiInputLabel-required-496 MuiInputLabel-root-492 MuiInputLabel-formControl-497 MuiInputLabel-animated-500"
                data-shrink={false}
              >
                Password
                <span
                  className="MuiFormLabel-asterisk-509"
                >
                   *
                </span>
              </label>
              <div
                className="MuiInputBase-root-523 MuiInput-root-510 MuiInput-underline-514 MuiInputBase-fullWidth-532 MuiInput-fullWidth-517 MuiInputBase-formControl-524 MuiInput-formControl-511"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="current-password"
                  className="MuiInputBase-input-533 MuiInput-input-518 MuiInputBase-inputType-536 MuiInput-inputType-521"
                  disabled={false}
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Type your password..."
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
            <button
              className="MuiButtonBase-root-482 MuiButton-root-540 MuiButton-contained-551 MuiButton-containedPrimary-552 MuiButton-raised-554 MuiButton-raisedPrimary-555 MuiButton-fullWidth-565"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="submit"
            >
              <span
                className="MuiButton-label-541"
              >
                Login
              </span>
              <span
                className="MuiTouchRipple-root-566"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
]
`;
